### PR TITLE
Filter of numbers

### PR DESCRIPTION
**Pull request by Sofasmile**

### CODE REVIEW

File: src/main/java/core/basesyntax/FilterOfNumber.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+public class FilterOfNumber {
+
+    /**
+     * Реализуйте метод filterMultiplesOfNumber(int[] array, int multiplier),
+     *      принимающий в качестве аргумента массив целых чисел `array`
```
Comment (roman-mate) [inline src/main/java/core/basesyntax/FilterOfNumber.java]:
```suggestion
     * принимающий в качестве аргумента массив целых чисел `array`
```

File: src/test/java/core/basesyntax/FilterOfNumberTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Random;
+
+public class FilterOfNumberTest {
+    @Test(expected = ArithmeticException.class)
+    public void checkDivisionRemainder() {
+        Random random = new Random();
```
Comment (roman-mate) [inline src/test/java/core/basesyntax/FilterOfNumberTest.java]: it's a bad practice to use random data in tests. If there is some edge case which is failing, test starts failing once in 20-50 runs and it's really hard to find a pattern. Those tests are called flaky. Flaky tests are worse than no tests, because you stop trusting your tests. Let's hardcode some data instead of generating random data.

File: src/test/java/core/basesyntax/FilterOfNumberTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+            int[] actualResult = checkRemainder.filterMultiplesOfNumber(array, value);
+            int[] expectedResult = correctFilterMultiplesOfNumber(array, value);
+            Assert.assertArrayEquals("Test failed with array: " + printArray(actualResult)
+                    + " exp: " + printArray(expectedResult)
+                    + " and value: " + value, actualResult, expectedResult);
+        }
+    }
+
+    private String printArray(int[] array) {
```
Comment (roman-mate) [inline src/test/java/core/basesyntax/FilterOfNumberTest.java]: I believe array.toString() returns human readable representation, doesn't it?

**Review by roman-mate — CHANGES_REQUESTED** [review]

**Review by roman-mate — APPROVED** [review]
